### DISCUSSION TITLE
기초 프로젝트 2주차 팀회고

### DISCUSSION THREAD

**Discussion by sulyoungoh** [codeit-bootcamp-frontend/3-Weekly-Mission#245] [회고]
어느덧 다음 주에 3기 기초 프로젝트 발표가 있습니다.
첫 협업이었던 만큼 많은 우여곡절이 있었을 텐데요,
흘러가는 시간을 회고를 통해 기록하며 치열했던 나와 팀원들의 모습을 기억하시길 바랄게요🥰

- 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
  - Keep : 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
  - Problem : 현재 우리팀이 가지고 있는 문제
  - Try : Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
- [KPT 회고 피그마 템플릿](https://www.figma.com/file/m40ny4qg670CPhP9PX4Pcb/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF_3%EA%B8%B0?type=whiteboard&node-id=0%3A1&t=Pk04kJlW4hOre6Cg-1)을 활용해서, 팀 회고를 진행하셔도 됩니다.
  - 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요.
- 이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.

**Comment by SiWooJinSeok** [comment]
## 6팀

## Keep
- 이번주도 역시 한명도 빠짐없이 협조하는 분위기여서 프로젝트 진행이 수월해서 좋았다.
- 기호님 의견으로 ‘코어타임’ 개념으로 다같이 집중 할 수 있는 시간 만들어진거 좋은듯
- 다들 충분히 잘 해주고 있고, 코드리뷰를 다들 해주신다.
- 새로운 것에 거부감 없이 잘 받아들인다.

## Problem
- 리팩토링 할게 너무 많다
- 스토리북이 어렵다.
- 남자가 4명이라 꾸미는게 어렵다  **ㅇㅈ**
- PR 템플릿을 정하지 않고 진행하다 보니 PR 양식이 제각각이라 가독성이 떨어진다. 또한 PR시 체크 하는 부분이 없다보니, 후반부에 다가오니 세세하게 컨벤션 정하지 않았던 부분들이 조금씩 드러나는 느낌.

## Try
- 끝까지 포기하지 않고 하기!
- 시간적 여유가 있으면 리펙토링하기
- 노오오오력을 한다
- PR 템플릿을 만들자!

**Comment by ohddang** [comment]
# 🌼Keep
피드백이 빨랐다.
PR 올라왔을 때 코드 리뷰를 성실하게 했다.
다들 성격이 좋아서 부딪힌 게 없다.
프로젝트를 진행하면서 다른 팀원이 작성한 코드를 보면서 몰랐던 것을 많이 배웠다.
코드를 이해할 수 있도록 주석을 열심히 달았다. (Todo, Info, Fix 등)

# 🛸Problem
완벽주의 성향이 짙어서 PR을 더 빨리 올리지 못했다.
지난 주에 Jira 이슈를 더 세분화하기로 했었는데 지키지 못했다.

# 💫Try
PR을 더 세분화해서 더 많이 자주 올리면 좋겠다.
커밋 메시지 양식을 더 정확하게 정하는 것이 좋겠다.

**Comment by soyxxn** [comment]
2팀
### 🥰 Keep

- 최대한 시간을 들여서 밤 늦은 시간까지 최선을 다하는 팀원들의 모습이 동기부여가 많이 됐다 !
- 완성한 것에 그치지 않고 프로젝트에 대해 다양한 의견을 나누고 계속해서 추가할 기능을 찾으려고 하는 모습이 보기 좋았다 !
- 문제가 생기면 포기하지 않고 끝까지 해결하려고 노력하는 모습 !
- 또 그 문제에 대해 다같이 해결하고자 노력하는 모습 !
- 연락이 잘 돼서 좋았다 !

### 🧐 Problem

- 공통 컴포넌트를 만들 때 초기에 조금 더 자주 소통하면서 만들었으면 좋았을 것 같다
- 개인의 형식으로 폴더구조나 파일명을 만들다보니 통일이 되지 않는 점
- 개인 파트에서는 소통을 제대로 못 한 것 같아 아쉽다
- PR을 올리면 코드리뷰를 많이 하지 않은 점
- UI와 비즈니스로직을 분리하는 부분
- 코드의 문제를 찾지 못해 불필요한 시간을 너무 많이 들였다

### 🤩 Try

- 더 많은 공부가 필요한 것 같다
- 프로젝트 마무리 단계에서 비효율적인 부분을 리팩토링하고 파일명이나 구조를 통일하면 좋을 것 같다
- 남은 시간동안 전체적인 스타일이나 용도별로 파일을 잘 분리하는 작업을 해보면 좋겠다

**Comment by 2zzzyoung** [comment]
## 3팀

### Keep
- git merge가 잘 되고 있는 건지는 모르겠지만 commit하고, push하고, pull하고, reabse merge하는 그 단계가 잘 유지되고 있는거 같다.
- 소통이 잘 이루어지는것같아 좋다. 제안을 하고, 그것을 해결하려고 노력하는 모습이 좋았다.
- 다들 연락이 잘 돼서 소통이 잘 되었다.
- 프로젝트를 하면서 큰 어려움 없이 스스로 잘 하는 것 같아 좋았다.
- 회의를 통해서 의사소통하는 것은 다들 프로젝트를 진행하면서 흐름을 알 수 있어서 좋은 것 같다. 
### Problem
- 아직은 익숙하지 않아서 불필요한 부분에 대한 삭제에 대해서 미처 얘기를 나누지 못한 점이 아쉽다. 
- 제 속도가 느린 것 같습니다. 열심히 하고 있긴 하지만 제 실력이 부족해서 나오는 것이라고 생각합니다.
- 시간이 부족하다 보니 머지하기 전에 코드 리뷰를 하지 못해 아쉬웠다.
- 아직은 코드의 일관성이 떨어진다 느껴지고 팀원 모두 깃에 대해 이해도가 좀 더 필요해보인다.
- 회의를 하는것은 좋지만 작은 것들도 자잘하게 회의를 진행해서 피로감을 느낀 것 같다.
### Try
- git의 merge, rebase를 좀 더 익혀서 문제없이 사용할 수 있도록, 문제가 발생해도 해결할 수 있도록 해야 될거 같다.
- 팀원분들께 막히는 부분에서 좀 여쭤봐야 할 것 같습니다. 급하게 강의를 듣는다고 해결되지는 않더라구요. 또한 다른 분들의 코드를 보면서 학습하도록 하겠습니다.
- 다른 사람의 코드를 보며 일관성을 맞춰가도록 노력해야 할 것 같다.
- 팀원의 PR이나 Merge된 코드를 한 번씩은 들여다보며 어떤 식으로 서로 코드를 작성해나가고 있는지 보면 좋을 것 같다.
- 초반을 지나고 나서 중간 회의를 함에 있어서 안건의 중요도가 어느정도 있는 것들이 있을 때 위주로 회의를 진행해보면 어떨까 싶다.

**Comment by jihee1103** [comment]
### 8팀
### Keep

- 팀회의에 빠지지 않고 참석하는 끈기
- 새벽까지 문제해결을 위해 포기하지 않는 팀원분들
- 서로서로 필요한 커뮤니케이션을 잘 했던 것 같다.

### Problem

- 개인적인 회고를 꼼꼼하게 잘 못썼었다.
- 처음부터 효율적으로 담당분배를 하지 못했다는 아쉬움.

### Try

- 자기가 작업하고있는 영역 분명히 하기
- 모르는 게 있으면 스스로 검색을 해보기
- 사전 질문을 할 때 문제사항, 시도한 점, 해결 방법으로 단계를 나누어서 하면 더 좋을 것 같습니다.
- 커뮤니케이션 적극적으로 취하기

**Comment by yuyoni** [comment]
## 👍 Keep

- 프로젝트 마무리가 다가올 때 더 느껴진 부분은 미리 프로젝트를 어떻게 진행할지 팀원들과 같이 논의했던 것이 좋았습니다. 덕분에 급하지도 않게 잘 마무리할 수 있을 것 같고, 더 꼼꼼하게 프로젝트를 마무리할 수 있을 것 같아 좋습니다.
- 다들 큰 갈등, 불만없이 본인이 맡은 역할은 잘 진행해주셔서 좋았습니다.
- 팀장인 유연님은 모두가 놓치는부분 캐치하거나 코드리뷰 열심히 해주셨고 작업도 항상 먼저 끝내놓으셔서 코드 작성하는데 도움이 많이됐습니다.
인덕님은 프로젝트 경험이 많으셔서 기본적인 세팅이나 컨벤션 조율하는데 큰역할을 하셨고 컨디션이 안좋으셨는데  가장 어려운 부분맡으셔서 깔끔하게 만들어주셨습니다.
재명님은 비슷한 컴포넌트 부분이 많았는데 먼저 작업하시고 공유해주셔서 큰도움이됐습니다. 코드리뷰도 항상 정성들여 해주셔서 감사합니다
선영님은 처음에 많이 힘들어하셨는데 포기안하시고 어떻게든 노력해서 해내시는게 보여서 좋았습니다.
모두고생하셨습니다.

## 💭 Problem

- 팀원들이 PR을 올리고 나서 컨펌하는 시간이 약간씩 오래 걸릴때가 있던 것 같습니다.
- 코드의 통일성이 없는 것 같고, 루트 폴더 구조를 미리 정해놓긴 했지만 하위폴더는 구체적으로 정하지 않아서 폴더구조 자체도 조금 알아보기 힘든 것 같습니다
- 새로운 라이브러리나 툴을 시도해보지 못한게 아쉽습니다. 새로운 기술도 공부해가며 프로젝트를 더 효율적으로 진행할 수 있었을 것 같은데 너무 배운 내용안에서 해결한 것 같아서 아쉽습니다.
- 다끝내고 보니 드는생각이지만 처음에 프로젝트 시작할때 코어타임에 디스코드에서 모여서 같이 작업했으면 능률이나 작업물 결과가 더 좋았지않을까싶네요.

## ⭐️ Try

- 팀원들이 많을수록 나중에 PR이 많이 쌓일 수도 있어서, PR을 올리면 컨펌과 코드 리뷰를 해서 빠르게 작업을 진행할 수 있게 하면 더 좋을 것 같습니다!!
- 처음에 컴포넌트나 공통 로직을 먼저 짠 후 큰 페이지로 넘어갈지, 각자 한 페이지를 맡을 지 고민하다 후자를 택했는데 시간이 조금 걸리고 복잡해도 전자가 훨씬 코드 통일성 및 재사용 방면에서는 좋았을 것 같다는 생각이 듭니다.
- 발표끝난후에 각자 맡은부분을 다시 분배해서 리뷰하거나 새롭게 만들어보면 좋을거같습니다.

**Reply by DanteSnow** [reply]
1팀

**Comment by DanteSnow** [comment]
## 5팀

---

### 💡**Keep**

- 주말인데도 모든 팀원들이 회의에 참석해서 긴 시간동안 서로 의견을 맞춰갔다.
- 목표하던 일정에 맞췄다.
- 프로젝트를 하면서 새로 접해본 api를 활용하여 기능 구현을 해보았다.
- 내가 짠 코드가 아닌 팀원들의 코드에서 작업을 해보았다.
- 팀 프로젝트를 하면서 회의 약속 시간을 잘 지켰다.
- 다른 팀원의 코드를 보면서 내 코드를 어떻게 수정하면 좋을지가 보이기 시작했다.

### ❓**Problem**

- 의견을 자유롭게 제시하고 이에 대해 토론하는 분위기가 형성되지 않은 것 같다.
- 1주차에 비해서 해이해졌던것같다.
- 처음 접하는 것이다 보니 기능 구현은 성공했지만 완벽하게 개념을 이해하지 못했다.
- 프로젝트 기한이 얼마 남지 않아서 팀원들의 코드를 제대로 확인하지 못했다.
- 내가 맡은 부분을 예정시간보다 빨리 끝냈을때 붕 뜨는 시간이 생겼다.

### 🏃**Try**

- 누군가 의견을 제시했고, 만약 그에 반 하는 의견이 있다면 모두가 충분히 납득할만한 이유와 대처방안을 제시하는 등의 건강한 토론 분위기가 형성되었으면 좋을 것 같다.
- 내가 얼마나 작업했는지 시간을 재가며 해야할것같다.
- 개발하면서 발생했던 문제들을 개념과 함께 정리해야겠다.
- 전체적인 흐름이라도 파악하려고 노력해야겠다.
- 가능한 팀이 정한 스케줄에 맞춰서 작업을 하고, 그래도 빨리 끝났을시 끝났다고 쉬지 말고 어떻게 개선 할 수 있을지 고민해보자.

**Comment by jiyong0106** [comment]
### 4팀

Keep

- 낮에도 밤에도 열심히 해서 완성은 아니지만 주요 기능들을 빠른 시간 안에 구현했다. 
- 궁금한 것, 사소한 부분들을 디스코드에 꾸준히 공유하며 소통을 했다. 
- 피그마에 정리된 기능들  외에도 능동적으로 생각해서 있으면 좋겠다! 하는 기능들을 만들었다.  
- 서로의 의견을 존중하고 좋은 분위기로 프로젝트를 마칠 수 있었다.  
- 깃허브 사용을 처음보다 능숙하게 할 수 있다.  
- 리모트 브랜치에서 pull 하고 conflict가 많이 없어서 시간을 아낄 수(?) 있었다.  

Problem

- 기능 구현에 집중하느라 pr을 merge용으로만 사용한 부분이 아쉽다. 
- 프로젝트가 거의 끝나다 보니 조금 긴장이 풀린 느낌 
- 전역 상태 관리 (Recoil) 를 하자는 의견은 있었지만 시도해보지 못했다. 
- Prop Drilling 덕분에 상태 관리의 중요성을 느꼈다. 

Try

- 발표자료 열심히 준비하기 
- 가독성이 좋도록 코드 리팩토링 하기
- 기회는 없지만 전역 상태 관리를 공부하고 도전하기
-  PR에 코드리뷰도 써보고 서로의 커밋을 확인하는 시간도 가져보기
